### PR TITLE
Add RoomChangePermissionsScreen

### PR DESCRIPTION
**Pull request by pixlwave**
First part of #2358 this PR adds the screen and theoretically the VM implementation, but it isn't part of a flow yet, so is untested against a real room.

Screenshot 2024-02-28 at 11 52 54 am

### CODE REVIEW

File: ElementX/Sources/Services/Room/RoomPermissions.swift [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                        invite: .user,
+                        kick: .moderator,
+                        redact: .moderator,
+                        eventsDefault: .user,
+                        stateDefault: .moderator,
+                        usersDefault: .user,
+                        roomName: .moderator,
+                        roomAvatar: .moderator,
+                        roomTopic: .moderator)
+    }
```
[the remark this answers is not in the document]
Comment (Velin92) [reply ElementX/Sources/Services/Room/RoomPermissions.swift:71]: Ofc this requires to create a permission Enum, which could actually also be used to turn the RoomPermissions (so `ban invite kick` and so on become part of an enum) from a struct into a dictionary (but up to you if you feel this is really worth doing)

[the remark this answers is not in the document]
Comment (pixlwave) [reply ElementX/Sources/Services/Room/RoomPermissions.swift:71]: If you don't mind, I'd like to leave this as is for now, and see what the defaults coming out of the SDK look like before figuring out the better path?

File: ElementX/Sources/Screens/RoomChangePermissionsScreen/RoomChangePermissionsScreenModels.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+extension RoomChangePermissionsScreenViewState {
+    /// Creates a view state for a particular group of permissions.
+    /// - Parameters:
+    ///   - currentPermissions: The current permissions for the room.
+    ///   - group: The group of permissions that should be shown in the screen.
+    init(currentPermissions: RoomPermissions, group: RoomRolesAndPermissionsScreenPermissionsGroup) {
+        switch group {
+        case .roomDetails:
+            // swiftlint:disable force_unwrapping
```
Comment (Velin92) [inline ElementX/Sources/Screens/RoomChangePermissionsScreen/RoomChangePermissionsScreenModels.swift:63]: check my other comment to see how to remove these unwrappings

**Review by Velin92 — CHANGES_REQUESTED** [review]
All good but I would prefer we could remove those force unwrappings, not for the risk but because they are ugly, and we can definitely get rid of them very easily

**Comment by pixlwave** [comment]
Alright, I've added a defaultValue method that shares the key path so it works in both locations. WDYT @Velin92?

**Review by Velin92 — APPROVED** [review]
LGTM

**Pull request merged**